integration/presto/src/main/prestodb/org/apache/carbondata/presto/readers/ComplexTypeStreamReader.java

```diff
@@ -139,7 +139,7 @@ public void putComplexObject(List<Integer> offsetVector) {
       Block rowBlock = RowBlock
           .fromFieldBlocks(childBlocks.get(0).getPositionCount(), Optional.empty(),
               childBlocks.toArray(new Block[0]));
-      for (int position = 0; position < childBlocks.get(0).getPositionCount(); position++) {
+      for (int position = 0; position < offsetVector.size(); position++) {
```

Comment:
missed this to handle at presto sql file, I wonder how the test case passed for prestosql

Reply:
We don't have that file in prestoDB profile

Reply:
please check agian. Both prestodb and prestosql has this class

integration/presto/src/main/prestosql/org/apache/carbondata/presto/readers/ComplexTypeStreamReader.java
integration/presto/src/main/prestodb/org/apache/carbondata/presto/readers/ComplexTypeStreamReader.java

Reply:
It had passed earlier because this is just refactoring. I have anyway added changes in other profile.